client/src/components/organisms/RequestGroupForm.tsx

```diff
@@ -0,0 +1,351 @@
+import { bindActionCreators, Dispatch } from "redux"
+import { convertFromRaw, EditorState } from 'draft-js';
+import { gql, useQuery } from "@apollo/client";
+import React, { FunctionComponent, useState } from "react";
+import { connect } from "react-redux";
+
+import AlertDialog from '../atoms/AlertDialog'
+import { Button } from '../atoms/Button'
+import FormItem from '../molecules/FormItem'
+import FormModal from './FormModal'
+import ImagePicker from '../atoms/ImagePicker'
+import RequestGroup from '../../data/types/requestGroup'
+import RichTextField from '../atoms/RichTextField'
+import { RootState } from '../../data/reducers'
+import { TagInput } from '../atoms/TagInput'
+import { TextField } from '../atoms/TextField'
+import { upsertRequestGroup } from '../../data/actions'
+
+interface StateProps {
+  requestGroups: Array<RequestGroup>
+}
+
+interface DispatchProps {
+  upsertRequestGroup: typeof upsertRequestGroup
+}
+
+
+type Props = StateProps
+  & DispatchProps
+  & {
+    handleClose: () => void,
+    requestGroupId?: string,
+    operation: "create" | "edit"
+  };
+
+const RequestGroupForm: FunctionComponent<Props> = (props: Props) => {
+  const [initialRequestGroup, setInitialRequestGroup] = useState<RequestGroup | null>(null)
+  const [showAlertDialog, setShowAlertDialog] = useState(false)
+  const [changeMade, setChangeMade] = useState(false)
+  const [name, setName] = useState("")
+  const [description, setDescription] = useState("")
+  const [image, setImage] = useState("")
+  const [requestTypesToIdMap, setRequestTypesToIdMap] = useState(new Map<string, string>())
+  const [requestTypeNames, setRequestTypeNames] = useState<Array<string>>([])
+  const [nameError, setNameError] = useState("")
+  const [descriptionError, setDescriptionError] = useState("")
+  const [imageError, setImageError] = useState("")
+  const [requestTypesError, setRequestTypesError] = useState("")
+
+  const images = ["https://source.unsplash.com/RcgiSN482VI",
```

Comment:
We need a better way to store this in the long-term. Something to keep in mind

Reply:
I think instead of hardcoding the list of images, I'll download from Unsplash, upload to a folder called request_images, then the codebase will just read all the files from the folder into an array

Reply:
@chamod-gamage by folder you mean on firebase?

Reply:
lol @chamod-gamage def avoid putting the pics in assets
bc when we offer photo upload, we'll have to go through their database and change the url of the images to copies in firebase storage

Reply:
Ya I mean fire base folder LOL

Reply:
LMFAO tyty for clarifying :)))